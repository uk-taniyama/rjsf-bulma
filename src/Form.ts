import type { ComponentType } from "react";

import { withTheme } from "@rjsf/core";

import Theme from "./Theme";

import type { FormProps } from "@rjsf/core";

const Form: ComponentType<FormProps> = withTheme(Theme);

export default Form;
