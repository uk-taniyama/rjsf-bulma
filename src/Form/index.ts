import type { FormProps } from "@rjsf/core";
import { withTheme } from "@rjsf/core";

import Theme from "../Theme";

const Form: React.ComponentType<FormProps> = withTheme(Theme);

export default Form;
