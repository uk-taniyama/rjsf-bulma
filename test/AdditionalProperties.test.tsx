import React from "react";

import validator from "@rjsf/validator-ajv6";
import renderer from "react-test-renderer";

import Form from "../src/index";

import type { RJSFSchema } from "@rjsf/utils";

describe("AdditionalProperties tests", () => {
  test("show add button and fields if additionalProperties is true", () => {
    const schema: RJSFSchema = {
      additionalProperties: true,
    };
    const formData: any = {
      additionalProperty: "should appear",
    };
    const tree = renderer
      .create(
        <Form schema={schema} validator={validator} formData={formData} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
