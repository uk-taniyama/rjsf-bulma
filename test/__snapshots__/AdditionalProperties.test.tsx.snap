// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalProperties tests show add button and fields if additionalProperties is true 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group field field-undefined"
  >
    <div
      className="field"
    >
      <div
        className="columns is-gapless"
      >
        <div
          className="column"
        >
           
          <div
            className="form-group field field-string additional-property"
          >
            <div
              className="columns is-gapless"
            >
              <div
                className="column is-3"
              >
                <div
                  className="field"
                >
                  <label
                    className="label"
                    htmlFor="root-key"
                  >
                    Key
                  </label>
                  <div
                    className="control"
                  >
                    <input
                      className="input"
                      defaultValue="additionalProperty"
                      disabled={false}
                      id="root-key"
                      name="root-key"
                      onBlur={[Function]}
                      readOnly={false}
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="column"
              >
                <div
                  className="field"
                >
                  <label
                    className="label"
                    htmlFor="root"
                  >
                    additionalProperty
                  </label>
                  <div
                    className="control"
                  >
                    <input
                      autoFocus={false}
                      className="input"
                      disabled={false}
                      id="root"
                      name="root"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      readOnly={false}
                      required={false}
                      type="text"
                      value="should appear"
                    />
                  </div>
                </div>
              </div>
              <div
                className="column is-narrow"
              >
                <button
                  className="button is-danger"
                  disabled={false}
                  onClick={[Function]}
                  title="Remove"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="columns is-gapless"
      >
        <div
          className="column"
        />
        <div
          className="column is-narrow"
        >
          <button
            className="button object-property-expand is-info"
            disabled={false}
            onClick={[Function]}
            title="Add Item"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <button
    className="button is-primary"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
